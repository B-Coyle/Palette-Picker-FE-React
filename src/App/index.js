import React, {Component} from 'react';
import './App.scss';

class App extends Component {
  constructor() {
    super()
    this.state = {
      project: [],
      loading: true,
      error: ''
    }
  }
  componentDidMount() {
  }


  render () {
    return(
      <h1>Palette Picker</h1>
      )
  }
}

export default App;
