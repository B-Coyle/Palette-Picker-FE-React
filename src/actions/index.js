export const hasError = message => ({
  type: "HAS_ERROR",
  message
});

export const saveProjects = projects => ({
  type: "SAVE_PROJECTS",
  projects
});

export const savePalette = palette => ({
  type: "SAVE_PALETTE",
  palette
});

export const addPalette = newPalette => ({
  type: "ADD_PALETTE",
  newPalette
});

export const addProject = newProject => ({
  type: "ADD_PROJECT",
  newProject
});

export const isLoading = boolean => ({
  type: "IS_LOADING",
  isLoading: boolean
});

export const deleteProject = (id) => ({
  type:"DELETE_PROJECT",
  id
})

export const deletePalette = (id)=> ({
  type:"DELETE_PALETTE",
  id
})