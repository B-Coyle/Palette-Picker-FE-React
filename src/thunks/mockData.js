export const mockPalettes = [
  {
    id: 1,
    palette_name: "Mountain Spring",
    color1: "White",
    color2: "Red",
    color3: "Blue",
    color4: "Green",
    color5: "Pink",
    project_id: 1
  },
  {
    id: 1,
    palette_name: "Mountain Fresh",
    color1: "Purple",
    color2: "Silver",
    color3: "Forest Green",
    color4: "White",
    color5: "Blue",
    project_id: 1
  }
];

export const mockProjects = [
    {
        id: 1,
        project_name: 'First Project',  
    }
];
