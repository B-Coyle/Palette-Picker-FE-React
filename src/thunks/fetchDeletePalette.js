import { hasError, isLoading, deletePalette } from "../actions";

export const fetchDeletePalette = id => {
  return async dispatch => {
    const url = `https://palette-picker-jbbc.herokuapp.com/api/v1/palettes/${id}`;
    const options = {
      method: "DELETE"
    };
    try {
      dispatch(isLoading(true));
      const response = await fetch(url, options);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(deletePalette(id));
      dispatch(isLoading(false));
    } catch (error) {
      dispatch(hasError(error.message));
    }
  };
};
