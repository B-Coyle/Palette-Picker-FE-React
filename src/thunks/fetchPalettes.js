import {hasError, savePalettes, isLoading} from '../actions';

export const fetchPalettes = () => {
    return async(dispatch) => {
        const url = ``;
        //need url here
        try {
            dispatch(isLoading(true))
            const response = await fetch(url)
            if(!response.ok) {
                throw Error (response.statusText)
            }
            const palettes = await response.json();
            dispatch(savePalettes(palettes))
            dispatch(isLoading(false))
        } catch (error) {
            dispatch(hasError(error.message))
        }
    }
}