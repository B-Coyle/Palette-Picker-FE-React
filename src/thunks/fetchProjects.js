import { hasError, saveProjects, isLoading } from '../actions';

export const fetchProjects = () => {
    console.log('FETCH PROJECTS');
    console.log(process.env);
    return async(dispatch) => {
        const url = `${process.env.REACT_APP_BACKEND_URL}/api/v1/projects`;
        console.log(url);
        try {
            dispatch(isLoading(true))
            const response = await fetch(url)
            if(!response.ok) {
                throw Error (response.statusText)
            }
            const projects = await response.json();
            console.log(projects);
            dispatch(saveProjects(projects))
            dispatch(isLoading(false))
        } catch(error){
            dispatch(hasError(error.message))
        }
    }
}