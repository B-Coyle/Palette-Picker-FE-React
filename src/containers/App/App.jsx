import React, {Component} from 'react';
import {connect} from 'react-redux';
import {fetchPalettes} from '../../thunks/fetchPalettes.js'
import './App.scss';

class App extends Component {
  constructor() {
    super()
    this.state = {
      projects: [],
      loading: true,
      error: ''
    }
  }
  componentDidMount() {
    this.props.fetchProjects();
    this.props.fetchPalettes();
  }


  render () {
    return(
      <main>
      <header>
        <h1>Palette Picker</h1>
      </header>
      <section class="palette-container">
        <div id="color1" class="color-container" data-locked={false}>
          <button id="1" class="color-lock" value='Lock Color'><i id="1" class="fas color-lock1 fa-lock-open"></i>
          </button>
          <p id="color1-name" class="color-hex"></p>
        </div>
        <div id="color2" class="color-container" data-locked={false}>
          <button id="2" class="color-lock" value='Lock Color'><i id="2" class="fas color-lock2 fa-lock-open"></i></button>
          <p id="color2-name" class="color-hex"></p>
        </div>
        <div id="color3" class="color-container" data-locked={false}>
          <button id="3" class="color-lock" value='Lock Color'><i id="3" class="fas color-lock3 fa-lock-open"></i></button>
          <p id="color3-name" class="color-hex"></p>
        </div>
        <div id="color4" class="color-container" data-locked={false}>
          <button id="4" class="color-lock" value='Lock Color'><i id="4" class="fas color-lock4 fa-lock-open"></i></button>
          <p id="color4-name" class="color-hex"></p>
        </div>
        <div id="color5" class="color-container" data-locked={false}>
          <button id="5" class="color-lock" value='Lock Color'><i  id="5" class="fas color-lock5 fa-lock-open"></i></button>
          <p id="color5-name" class="color-hex"></p>
        </div>
      </section>
      <section class='generate-palette'>
          <input type='button' class='generate-palette-btn' value='Regenerate Colors' />
      </section>
      <section class='form-container'>
        <form id='name-project-form'>
          <p class="project-exists hidden">Project already exists</p>
          <input class='project-name-input' type='text' placeholder="New Project Name" />
          <input class='button' id='create-project-btn' type="button" value='Create Project' />
        </form> 
      </section>
      <section class='form-container'>
        <form id='name-palette-form'>
          <select id='project-select'><option value='' disabled>Please Select A Project</option> */}
          </select>
          <input class='palette-name-input' type='text' placeholder='Palette Name' />
          <input class='button' type="button" id='create-palette-btn' value='Create Palette' />
        </form>
      </section>
      <section class='saved-projects-section'>
        <div class='saved-projects-container'></div>
      </section>
    </main>
      )
  }
}

export default App;
