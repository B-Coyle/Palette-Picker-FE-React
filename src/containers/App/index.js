import React, {Component} from 'react';
import {connect} from 'react-redux';
import {fetchPalettes} from '../../thunks/fetchPalettes.js'
import './App.scss';

class App extends Component {
  constructor() {
    super()
    this.state = {
      project: [],
      loading: true,
      error: ''
    }
  }
  componentDidMount() {
    this.props.fetchProjects();
    this.props.fetchPalettes();
  }


  render () {
    return(
      <h1>Palette Picker</h1>
      )
  }
}

export default App;
