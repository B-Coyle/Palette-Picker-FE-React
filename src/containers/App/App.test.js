import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps } from './App';
import { fetchProjects } from '../../thunks/fetchProjects';
import { fetchPalettes } from '../../thunks/fetchPalettes';
import { mockProjects } from '../../utils/mockData';
import * as actions from '../../actions';
import { isMainThread } from 'worker_threads';

jest.mock('../../thunks/fetchProjects')
jest.mock('../../thunks/fetchPalettes')


describe('app', () => { 
  let wrapper 
  let fetchProjectsMock = jest.fn()
  let fetchPalettesMock = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <App
        fetchPalettes={fetchPalettesMock}
        fetchProjects={fetchProjectsMock}
      />
    );
  });

  it('should render the component elements', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when fetchProjects is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = fetchProjects();
      const mappedProperties = mapDispatchToProps(mockDispatch)
      mappedProperties.fetchProjects();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it("should call dispatch when fetchPalettes is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchPalettes();
      const mappedProperties = mapDispatchToProps(mockDispatch);
      mappedProperties.fetchPalettes();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
