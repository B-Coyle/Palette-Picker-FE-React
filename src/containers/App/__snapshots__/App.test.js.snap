// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render the component elements 1`] = `ShallowWrapper {}`;
